pandas/tests/arithmetic/test_numeric.py

```diff
@@ -74,33 +75,22 @@ def test_compare_invalid(self):
 
 # ------------------------------------------------------------------
 # Numeric dtypes Arithmetic with Datetime/Timedelta Scalar
+index_or_series_params = [
+    pd.Series,
+    pd.Index,
+]  # type: List[Union[Type[pd.Index], Type[pd.RangeIndex], Type[pd.Series]]]
+left = [pd.RangeIndex(10, 40, 10)]  # type: List[Union[Index, Series]]
+for cls in index_or_series_params:
+    for dtype in ["i1", "i2", "i4", "i8", "u1", "u2", "u4", "u8", "f2", "f4", "f8"]:
+        left.append(cls([10, 20, 30], dtype=dtype))
 
 
 class TestNumericArraylikeArithmeticWithDatetimeLike:
 
     # TODO: also check name retentention
     @pytest.mark.parametrize("box_cls", [np.array, pd.Index, pd.Series])
     @pytest.mark.parametrize(
-        "left",
-        [pd.RangeIndex(10, 40, 10)]
-        + [
```

Comment:
I imagine you were getting an error when adding lists together. This is an unfortunate limitation of mypy that you cannot use the `+` operator and get a Union of List types.

If you wanted to minimize diff you could alternately just unpack the second argument, so `[[pd.RangeIndex(10, 40 10)], *[cls([10, 20, 30], dtype=dtype) for dtype in [...`

mypy history goes back to:

https://github.com/python/typeshed/pull/2653

Definitely unfortunate

Reply:
Yep, that's exactly what I hit.